cli/internal/cmd/verify.go

```diff
@@ -207,6 +233,139 @@ func addPortIfMissing(endpoint string, defaultPort int) (string, error) {
 	return "", err
 }
 
+// a formatter formats the attestation document.
+type formatter interface {
+	// format returns the raw or formatted attestation doc depending on the rawOutput argument.
+	format(docString string, rawOutput bool, expectedPCRs measurements.M) (string, error)
+}
+
+type attestationDocFormatter struct {
+	log debugLog
+}
+
+// format returns the raw or formatted attestation doc depending on the rawOutput argument.
+func (f *attestationDocFormatter) format(docString string, rawOutput bool, expectedPCRs measurements.M) (string, error) {
+	b := &strings.Builder{}
+	b.WriteString("Attestation Document:\n")
+	if rawOutput {
+		b.WriteString(fmt.Sprintf("%s\n", docString))
+		return b.String(), nil
+	}
+
+	var doc attestationDoc
+	if err := json.Unmarshal([]byte(docString), &doc); err != nil {
+		return "", fmt.Errorf("unmarshal attestation document: %w", err)
+	}
+
+	if err := f.parseQuotes(b, doc.Attestation.Quotes, expectedPCRs); err != nil {
+		return "", fmt.Errorf("parse quote: %w", err)
+	}
+
+	instanceInfoString, err := base64.StdEncoding.DecodeString(doc.InstanceInfo)
+	if err != nil {
+		return "", fmt.Errorf("decode instance info: %w", err)
+	}
+
+	var instanceInfo azureInstanceInfo
+	if err := json.Unmarshal(instanceInfoString, &instanceInfo); err != nil {
+		return "", fmt.Errorf("unmarshal instance info: %w", err)
+	}
+
+	if err := f.parseCerts(b, "VCEK certificate", instanceInfo.Vcek); err != nil {
+		return "", fmt.Errorf("print VCEK certificate: %w", err)
+	}
+	if err := f.parseCerts(b, "Certificate chain", instanceInfo.CertChain); err != nil {
+		return "", fmt.Errorf("print certificate chain: %w", err)
+	}
+
+	return b.String(), nil
+}
+
+// parseCerts parses the base64-encoded PEM certificates and writes their details to the output builder.
+func (f *attestationDocFormatter) parseCerts(b *strings.Builder, certTypeName string, encCertString string) error {
+	certBytes, err := base64.StdEncoding.DecodeString(encCertString)
+	if err != nil {
+		return fmt.Errorf("decode %s: %w", certTypeName, err)
+	}
+	formattedCert := strings.ReplaceAll(string(certBytes[:len(certBytes)-1]), "\n", "\n\t\t") + "\n"
+	b.WriteString(fmt.Sprintf("\tRaw %s:\n\t\t%s", certTypeName, formattedCert))
+
+	f.log.Debugf("Decoding PEM certificate: %s", certTypeName)
+	i := 1
+	block, rest := pem.Decode(certBytes)
+	for block != nil {
```

Comment:
```suggestion
	for block, rest := pem.Decode(certBytes); block != nil {
```

Reply:
I think this will set the rest and block for every loop iteration, which is not the behavior we want. If we would do this, we would have to set certBytes at the end of the loop body which seems a little counter-intuitive to me. What do you think?